org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/plain/AddJarsCommand.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.jdt.ls.extension.core.internal.plain;
+
+import static java.lang.String.format;
+import static org.eclipse.jdt.core.JavaCore.newLibraryEntry;
+import static org.eclipse.jdt.ls.core.internal.JDTUtils.PATH_SEPARATOR;
+
+import com.google.common.base.Preconditions;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.eclipse.che.jdt.ls.extension.core.internal.JavaModelUtil;
+import org.eclipse.core.resources.IContainer;
+import org.eclipse.core.resources.IFolder;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.IClasspathEntry;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin;
+
+/**
+ * Adds all jars from library folder into project's classpath.
+ *
+ * @author Valeriy Svydenko
+ */
+public class AddJarsCommand {
+  /**
+   * Adds jars into classpath.
+   *
+   * @param arguments contains two arguments: project uri and library folder's name
+   * @param pm a progress monitor
+   * @return uri of the project
+   */
+  public static Object execute(List<Object> arguments, IProgressMonitor pm) {
+    Preconditions.checkArgument(
+        arguments.size() >= 2, "Project uri and library folder are expected");
+
+    final String projectUri = (String) arguments.get(0);
+    final String libFolder = (String) arguments.get(1);
+
+    String projectName = projectUri.substring(projectUri.lastIndexOf(PATH_SEPARATOR) + 1);
+
+    IProject project = ResourcesPlugin.getWorkspace().getRoot().getProject(projectName);
```

Comment:
This is both unnecessary and wrong. The project name does not have to be the same as the name of the project folder. When looking up resources, we should always use the URI. Also, we're already checking for existence a java project below and fail with the same error.